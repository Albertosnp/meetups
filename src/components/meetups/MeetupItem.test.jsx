/* eslint-disable testing-library/no-debugging-utils */
import { fireEvent, render, screen, configure } from '@testing-library/react';
import ReactDOM from 'react-dom';
import MeetupItem from './MeetupItem';

configure({ testIdAttribute: 'data-cy' });

describe('Meetup-Item Card', () => {
  const defaultProps = {
    image: 'https://via.placeholder.com/150',
    title: 'new meetup',
    address: 'new meetup adress',
    description: 'this is a new meetup',
    handleFavoriteClick: jest.fn()
  };

  const getComponent = props => <MeetupItem {...props} />;

  const setup = (props = defaultProps) => render(getComponent(props));

  it('Should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(getComponent(), div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('Should match snapshot with full props', () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it('should has properties indicated', () => {
    setup();

    screen.getByText('new meetup');
    screen.getByText('new meetup adress');
    screen.getByText('this is a new meetup');
  });

  it('clicking the button calls event like handle once', () => {
    const mockHandle = jest.fn();
    setup({ ...defaultProps, handleFavoriteClick: mockHandle });

    const button = screen.getByTestId('add-favourite');
    fireEvent.click(button);

    expect(mockHandle).toHaveBeenCalledTimes(1);
    expect(mockHandle).not.toHaveBeenCalledTimes(3);
  });
});
