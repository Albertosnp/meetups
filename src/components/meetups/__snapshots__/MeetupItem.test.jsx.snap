// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Meetup-Item Card Should match snapshot with full props 1`] = `
<div>
  <li
    class="item"
    data-test="meet-up-item"
  >
    <div
      class="card"
    >
      <div
        class="image"
      >
        <img
          alt="new meetup"
          src="https://via.placeholder.com/150"
        />
      </div>
      <div
        class="content"
      >
        <h3>
          new meetup
        </h3>
        <address>
          new meetup adress
        </address>
        <p>
          this is a new meetup
        </p>
      </div>
      <div
        class="actions"
      >
        <button
          data-cy="add-favourite"
        >
          Add to favorites
        </button>
      </div>
    </div>
  </li>
</div>
`;
